tests/RequestMatcher/ClosureRequestMatcher.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Kevinrob\GuzzleCache\Tests\RequestMatcher;
+
+use Closure;
+use Kevinrob\GuzzleCache\Strategy\Delegate\RequestMatcherInterface;
+use Psr\Http\Message\RequestInterface;
+
+class ClosureRequestMatcher implements RequestMatcherInterface
+{
+    /**
+     * @var Closure
+     */
+    private $closure;
+
+    /**
+     * CallbackRequestMatcher constructor.
+     */
+    public function __construct(Closure $closure)
```

Comment:
Well, the `Kevinrob\GuzzleCache\Tests\RequestMatcher\ClosureRequestMatcher` class is actually only intended to unit tests (as its namespace implies). If PHP7 were the minimum requirement, I would have used an anonymous class for using it in PHPUnit. As a consequence, since this class should not be used outside of `Kevinrob\GuzzleCache\Tests\DelegatingCacheTest`, another type-hinting is pointless.

This class is indeed just an example of implementation of `RequestMatcherInterface`, but I don't think this example should be followed for production: since request matching targets specific use cases, I suggest letting developers implement `RequestMatcherInterface` by themselves instead of providing a default class. What's your opinion?

If you think a default implementation should be provided, then I suggest the removal of the `RequestMatcherInterface` and type-hint a `callable` in `DelegatingCacheStrategy::registerRequestMatcher()`, that should return a boolean.

What do you think?

Besides, since the default PHP version is 5.5 (so, no return type-hinting), do you think we should throw an `InvalidArgumentException` when the request matcher does not return a true `boolean`? Or do we do this the lazy way?